crates/holochain_zome_types/src/init.rs

```diff
@@ -1,26 +1,25 @@
-use crate::zome_io::ExternIO;
 use crate::CallbackResult;
-use holo_hash::EntryHash;
+use holo_hash::AnyDhtHash;
 use holochain_serialized_bytes::prelude::*;
+use holochain_wasmer_common::WasmError;
 
 #[derive(Clone, PartialEq, Serialize, Deserialize, SerializedBytes, Debug)]
 pub enum InitCallbackResult {
     Pass,
     Fail(String),
-    UnresolvedDependencies(Vec<EntryHash>),
-}
-
-impl From<ExternIO> for InitCallbackResult {
-    fn from(callback_guest_output: ExternIO) -> Self {
-        match callback_guest_output.decode() {
-            Ok(v) => v,
-            Err(e) => Self::Fail(format!("{:?}", e)),
-        }
-    }
+    UnresolvedDependencies(Vec<AnyDhtHash>),
 }
 
 impl CallbackResult for InitCallbackResult {
     fn is_definitive(&self) -> bool {
         matches!(self, InitCallbackResult::Fail(_))
     }
+    fn try_from_wasm_error(wasm_error: WasmError) -> Result<Self, WasmError> {
+        match wasm_error {
+            WasmError::Guest(_) | WasmError::Serialize(_) | WasmError::Deserialize(_) => {
+                Ok(InitCallbackResult::Fail(wasm_error.to_string()))
+            }
+            _ => Err(wasm_error),
```

Comment:
I wonder if we shouldn't just explicitly put in every variant here, i.e. maybe `WasmError` should be `#[non_exhaustive]`, since it's pretty important that this stays future proof?

If so, same goes for every other impl

Reply:
Or if that's too boilerplatey we could wrap this in a function that takes a `Fn(WasmError) -> T` since it seems to be these three variants all the time. Except when it's not.